core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/service/GitVersionControlLocalFileStorageService.java

```diff
@@ -0,0 +1,238 @@
+package edu.uci.ics.texera.web.resource.dashboard.user.dataset.service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+/**
+ * Git-based implementation of the VersionControlFileStorageService, using local file storage.
+ */
+public class GitVersionControlLocalFileStorageService {
```

Comment:
we may also consider remote FS in the future. check out #2326 and we can have a discussion.